### PR TITLE
[OpenAPI] OAuth

### PR DESCRIPTION
**Pull request by maartenvn**
Add OAuth endpoints to the OpenAPI spec.

**This is different than the current spec and should be updated**

Partially implements: #48

### CODE REVIEW

**Review by Jan-PieterBaert — COMMENTED** [review]
It looks good to me, but I'll let the people that actually write code for the backend review this

**Comment by mcbloch** [comment]
You don't want a callback with the provider also in the url??

**Comment by maartenvn** [comment]
> You don't want a callback with the provider also in the url??

If you are talking about the `{provider}` part in the url, thats just as a placeholder. I didn't want to write the same spec for different providers.

**Comment by mcbloch** [comment]
Yeah but in the callback url you left out the provider meaning for ex. github and zeus oauth will both end up at the`/oauth/callback` url. If that's intentional then it's okay but I was wondering if it was.

**Comment by maartenvn** [comment]
It was like that in the original backend. With a provider is fine as well, whatever you prefer.

**Comment by mcbloch** [comment]
We'll do it with. That way you can still choose to handle them differently if you want to.

File: openapi/reference/Default.v1.yaml [hunk 1]
```diff
@@ -885,5 +885,79 @@ paths:
...
+      summary: Your GET endpoint
+      tags: []
+      responses:
+        '307':
+          description: Temporary Redirect
+          headers: {}
+      operationId: get-oauth-zeus
+      description: |-
+        OAuth endpoint for login with {provider}.
+        This should redirect to the backend /oauth/callback after completion.
```
Comment (mcbloch) [inline openapi/reference/Default.v1.yaml]:
```suggestion
        This should redirect to the backend /oauth/{provider}/callback after completion.
```

File: openapi/reference/Default.v1.yaml [hunk 2]
```diff
@@ -885,5 +885,79 @@ paths:
...
+      - schema:
+          type: string
+          enum:
+            - zeus
+            - github
+        name: provider
+        in: path
+        required: true
+        description: 'OAuth provider (zeus, github, ...)'
+  /oauth/callback:
```
Comment (mcbloch) [inline openapi/reference/Default.v1.yaml:923]:
```suggestion
  /oauth/{provider}/callback:
```

**Review by mcbloch — CHANGES_REQUESTED** [review]

**Pull request merged**